### PR TITLE
Support named character references specified by the living standard

### PR DESCRIPTION
**Pull request by bptato**
https://duckduckgo.com uses it for example.

BTW, https://html.spec.whatwg.org/multipage/named-characters.html#named-character-references lists 2k+ entities, w3m supports 260 (+1 in this patch). Would a patch adding all entities in that list be welcome?

### CODE REVIEW

**Comment by tats** [comment]
> Would a patch adding all entities in that list be welcome?

Acceptable if getescapechar() really works.

Multiple character references may be unsupported.

**Comment by bptato** [comment]
Ok, let's repurpose this PR then.

> Multiple character references may be unsupported.

In the latest commit they are simply omitted. They could be supported if we stored entities as unsigned long (multiple character references here really just mean two codepoints instead of one), but then all places where getescapechar is used would have to be changed to take that into account.
Well, it's not used that much, so it should be possible.

Most specified non-terminated character references (i.e. without `;`) aren't encoded correctly either, but I'd rather not touch that part as I'm afraid of breaking URL query strings...

File: entity.js [hunk 1]
```diff
@@ -39,15 +39,15 @@ function gentestitem(b, c, d) {
 		b +
 		"</td>" +
 		"<td>" +
-		c.characters +
+		c.characters.replace('&', '&amp;').replace('<', '&lt;').replace('>', '&gt') +
```
Comment (tats) [inline entity.js]: Should `'&gt'` be `'&gt;'`?

Comment (bptato) [reply entity.js]: Technically it's the same but for consistency's sake yes.

**Comment by tats** [comment]
Merged, thanks for your contribution.

**Pull request merged**